### ISSUE TITLE
Can't open file gyp_uv.py

### ISSUE THREAD

**Issue by f0rb1dd3n** [libuv/libuv#1513]
I am using Windows 10 x64 with VS2017 and I have compiled x86, so I copied the libuv.lib, delete the folder, and clone again. When I tried to compile for x64, I got this error:

$ ./vcbuild.bat vs2017 x64
Looking for Visual Studio 2017
calling: "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\\Auxiliary\Build\vcvarsall.bat" x64
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26730.12
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
Found MSVS version 15.0
C:\Python27\python.exe: can't open file 'gyp_uv.py': [Errno 2] No such file or directory
Failed to create vc project files.

I dont know why it happens, I have tried a lot to figure out, but I am stuck. Can somebody help me?

**Comment by bzoz** [comment]
Is there gyp_uv.py in the folder?

**Comment by f0rb1dd3n** [comment]
Yeah, there is gyp_uv.py properly. I tried to execute with python --help just for test and it works.

I clone libuv again to try, and the same error occur

I have a question: Is there a release of libuv.lib (x86 and x64) compiled statically? Thanks.

**Comment by bzoz** [comment]
You do a clean clone of libuv and the build fails - is that correct?

I've also noticed you are using `$ ./vcbuild.bat vs2017 x64`, so I guess you do this from Git Bash. Did you try running the command from Windows shell?

**Comment by f0rb1dd3n** [comment]
yeah, I tried to run on a VS prompt shell. I will try another tests now and I will notice about.

**Comment by basilinjoe** [comment]
```
Looking for Visual Studio 2017
calling: "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\\Auxi
liary\Build\vcvarsall.bat" x86
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26730.10
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x86'
Found MSVS version 15.0
python: can't open file 'gyp_uv.py': [Errno 2] No such file or directory
Failed to create vc project files.
```
Same issue here

**Comment by f0rb1dd3n** [comment]
I could not figure out this error. So, I just restore a backup of a virtual machine and install all dependencies. But the virtual machine that is giving me this error still not working.

**Comment by carenas** [comment]
FWIW I might had found this same issue and worked it around accidentally by calling vcsbuild.bat from a regular command prompt (not the visual studio one)

**Comment by bzoz** [comment]
@f0rb1dd3n @basilinjoe The only think I can think off is current work directory being changed for whatever reason or `%PYTHON%` set to something incorrect. Could you add
```
cd
echo %PYTHON%
```
to `vcbuild.bat`, just before `"%PYTHON%" gyp_uv.py ...`? This should print current work directory and `%PYTHON%` variable.

**Comment by carenas** [comment]
this might be a duplicate of #1545, which has a proposed solution in #1546

**Comment by bzoz** [comment]
@f0rb1dd3n was this fixed by https://github.com/libuv/libuv/pull/1546?

**Comment by refack** [comment]
That's the error I used to see pre #1546.

tl;dr possibly caused by a `vcvarsall.bat` bug if a `%USERPROFILE%\source` directory exists

**Comment by f0rb1dd3n** [comment]
@bzoz, sorry my delay man. As I was testing on a VM, I just restore the backup after I install the module, and tried again. Then, the building works nice. I did not test #1546. 

Thanks your help man!

**Comment by bzoz** [comment]
I guess we can close this then. If the issue ever happens again feel free to reopen it.
